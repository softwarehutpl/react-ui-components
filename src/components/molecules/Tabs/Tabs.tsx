import React from 'react';

export const Tabs: React.StatelessComponent<{}> = () => (
  <div className="">Tabs
  </div>
);
