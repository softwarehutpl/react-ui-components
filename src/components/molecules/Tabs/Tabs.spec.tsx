import React from 'react';
import { create, act } from 'react-test-renderer';
import 'jest-styled-components';
import Tabs, { Wrapper, TabsWrapper } from './Tabs';

describe('<Tabs />', () => {
  it('should render corectly', () => {
    const component = create(<Tabs />);
    expect(component).toMatchSnapshot();
  });

  it('should render with className if provied', () => {
    const component = create(<Tabs className="test-class" />);
    const instance = component.root;
    instance.findByType('div');
    expect(instance.props.className).toEqual(expect.stringContaining('test-class'));
  });

  it('should have propery style with vertical orientation', () => {
    const component = create(<Wrapper orientation="vertical" />).toJSON();
    const tabsComponent = create(<TabsWrapper orientation="vertical" />).toJSON();
    expect(component).toHaveStyleRule('flex-direction', 'row');
    expect(tabsComponent).toHaveStyleRule('flex-direction', 'column');
  });

  it('should have proper style if margin props is provied', () => {
    const component = create(<Wrapper margin={15} />).toJSON();
    expect(component).toHaveStyleRule('margin', '15px');
  });
});
