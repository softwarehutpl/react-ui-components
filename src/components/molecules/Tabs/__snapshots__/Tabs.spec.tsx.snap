// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tabs /> should render corectly 1`] = `
.c0 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: space-between;
  -ms-flex-line-pack: space-between;
  align-content: space-between;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c2 {
  width: 100%;
}

<div
  className="c0"
  orientation="horizontal"
>
  <div
    className="c1"
    orientation="horizontal"
  />
  <div
    className="c2"
  />
</div>
`;
